lib/evmone/baseline.cpp

```diff
@@ -12,16 +12,21 @@ namespace evmone
 {
 JumpdestMap build_jumpdest_map(const uint8_t* code, size_t code_size)
 {
-    JumpdestMap m(code_size);
+    // Bitmask for PUSH instructions identification.
+    // It removes the lower 5 bits of a PUSH instruction with information about the data length.
+    // The remaining top 3 bits of every PUSH instruction is 0b011, i.e. 0x60, OP_PUSH1.
+    constexpr auto push_op_mask = 0b11100000;
+
+    JumpdestMap map(code_size);
     for (size_t i = 0; i < code_size; ++i)
     {
         const auto op = code[i];
-        if (op == OP_JUMPDEST)
-            m[i] = true;
-        else if (op >= OP_PUSH1 && op <= OP_PUSH32)
-            i += op - size_t{OP_PUSH1 - 1};
+        if ((op & push_op_mask) == OP_PUSH1)
```

Comment:
Nit: maybe simpler
```suggestion
        if ((op & push_op_mask) == push_op_mask)
```
or add `static_assert(push_op_mask == OP_PUSH1)`

Reply:
These are different. The mask is 0xE0.